### PR TITLE
LPS-162952 Make portal use the domain to determine the right cookie handling configuration

### PR DESCRIPTION
**Pull request by dsanz**
In this PR we're solving the issue analyzed [here](https://liferay.atlassian.net/wiki/spaces/ENGFRONTENDINFRA/pages/2164228128/LPS-165962+Research+cases+for+cross-site+navigations+under+the+same+domain+in+terms+of+cookie+handling), where we realized that visiting several liferay sites using the **same hostname** breaks the application of per-site cookie handling settings, producing some weird effects which are unnoticed by end users. Please refer to the analysis for details.

### Proposed solution 
Use the **hostname** of the incoming request to decide the scope of applicable cookie handling settings, as follows:
* If hostname is a declared instance-level virtual host, then use the corresponding instance settings, no matter if targeted site overrides them
* If hostname is a site-level virtual host, then use the corresponding site settings. If they’re not defined, then instance-level ones will be used instead
* Otherwise, use the default instance settings, no matter if targeted site overrides them

As a result, the cookie handling settings one gets while visiting a site depend on the hostname used to visit it. This provides a consistent experience with no unnoticed changes to the user, at the cost of making the site-level settings applicable only when the site-level virtualhost is used to browse the site content.

cc/ @jkappler @tinycarol I'm marking this as draft because I'd like to discuss some decisions I've made

### CODE REVIEW

File: modules/apps/cookies/cookies-banner-web/src/main/java/com/liferay/cookies/banner/web/internal/display/context/BaseCookiesBannerDisplayContext.java [hunk 1]
```diff
@@ -47,28 +49,43 @@ public BaseCookiesBannerDisplayContext(
...
 
-	public List<ConsentCookieType> getOptionalConsentCookieTypes(long groupId)
+	public List<ConsentCookieType> getOptionalConsentCookieTypes()
 		throws Exception {
 
 		if (_optionalConsentCookieTypes != null) {
 			return _optionalConsentCookieTypes;
 		}
 
-		_optionalConsentCookieTypes =
```
Comment (jkappler) [inline modules/apps/cookies/cookies-banner-web/src/main/java/com/liferay/cookies/banner/web/internal/display/context/BaseCookiesBannerDisplayContext.java:57]: @dsanz, with this change we can remove all methods in CookiesManagerUtil right? I noticed that we only use now add and get cookies

File: modules/apps/cookies/cookies-api/src/main/java/com/liferay/cookies/configuration/CookiesConfigurationHelper.java [hunk 2]
```diff
@@ -32,8 +30,7 @@ public CookiesConsentConfiguration getCookiesConsentConfiguration(
     ThemeDisplay themeDisplay) throws Exception;
 
 	public CookiesPreferenceHandlingConfiguration
-			getCookiesPreferenceHandlingConfiguration(
-				HttpServletRequest httpServletRequest)
```
Comment (jkappler) [inline modules/apps/cookies/cookies-api/src/main/java/com/liferay/cookies/configuration/CookiesConfigurationHelper.java]: Really like this simplification!

**Comment by jkappler** [comment]
@dsanz, really like this approach

File: modules/apps/cookies/cookies-banner-web/src/main/java/com/liferay/cookies/banner/web/internal/display/context/BaseCookiesBannerDisplayContext.java (hunk 1 above)
Comment (dsanz) [reply modules/apps/cookies/cookies-banner-web/src/main/java/com/liferay/cookies/banner/web/internal/display/context/BaseCookiesBannerDisplayContext.java:57]: Probably yes, I just took logic from the corresponding implementations from CookieManagerImpl to break the dependency with the `groupId` when preparing the context data. So the equivalent `getOptionalConsentCookieTypes()` and `getRequiredConsentCookieTypes()` in CookieManager might become unused

File: modules/apps/cookies/cookies-impl/src/main/java/com/liferay/cookies/internal/configuration/helper/CookiesConfigurationHelperImpl.java [hunk 3]
```diff
@@ -23,91 +23,66 @@
...
+	@Override
+	public CookiesBannerConfiguration getCookiesBannerConfiguration(
+			ThemeDisplay themeDisplay)
+		throws Exception {
 
-    if (layoutSet != null) {
-      Group group = layoutSet.getGroup();
+		return _getCookiesConfiguration(
+			CookiesBannerConfiguration.class, themeDisplay);
+	}
```
Comment (dsanz) [inline modules/apps/cookies/cookies-impl/src/main/java/com/liferay/cookies/internal/configuration/helper/CookiesConfigurationHelperImpl.java:44]: This should have been made in https://github.com/liferay-frontend/liferay-portal/pull/2846/commits/32c5d9befa99402eb5db2f90ac576a36917e1426

File: modules/apps/cookies/cookies-impl/src/main/java/com/liferay/cookies/internal/configuration/helper/CookiesConfigurationHelperImpl.java [hunk 4]
```diff
@@ -23,91 +23,66 @@
...
+			ThemeDisplay themeDisplay)
+		throws Exception {
 
-    return _configurationProvider.getCompanyConfiguration(
-      CookiesConsentConfiguration.class,
-      themeDisplay.getCompanyId());
-  }
+		return _getCookiesConfiguration(
+			CookiesConsentConfiguration.class, themeDisplay);
+	}
```
Comment (dsanz) [inline modules/apps/cookies/cookies-impl/src/main/java/com/liferay/cookies/internal/configuration/helper/CookiesConfigurationHelperImpl.java:53]: Likewise, this should have been made in https://github.com/liferay-frontend/liferay-portal/pull/2846/commits/32c5d9befa99402eb5db2f90ac576a36917e1426

**Comment by jkappler** [comment]
@dsanz, as we talked, just small changes. All looked good! 

I rebased with master

File: modules/apps/cookies/cookies-impl/src/main/java/com/liferay/cookies/internal/configuration/helper/CookiesConfigurationHelperImpl.java [hunk 5]
```diff
@@ -82,25 +82,27 @@ public CookiesConsentConfiguration getCookiesConsentConfiguration(
...
+        clazz, group.getGroupId());
+    }
 
-		return _configurationProvider.getCompanyConfiguration(
-			CookiesPreferenceHandlingConfiguration.class,
-			themeDisplay.getCompanyId());
-	}
+    return _configurationProvider.getCompanyConfiguration(
+      clazz, themeDisplay.getCompanyId());
+  }
```
Comment (dsanz) [inline modules/apps/cookies/cookies-impl/src/main/java/com/liferay/cookies/internal/configuration/helper/CookiesConfigurationHelperImpl.java]: Thanks for putting these changes together @jkappler :hugs:

**Comment by gaborlovasdotcom** [comment]
ci:test:sf

**Comment by gaborlovasdotcom** [comment]
ci:test:cookies

**Comment by gaborlovasdotcom** [comment]
@dsanz Added test fixes for existing functional tests.
Tests are passed see https://github.com/gaborlovasdotcom/liferay-portal/pull/449, [[master] ci:test:cookies - gaborlovasdotcom > gaborlovasdotcom - PR#449 - 2022-11-13[02:07:18]](https://testray.liferay.com/home/-/testray/runs?testrayBuildId=2051036818)

**Comment by tinycarol** [comment]
ci:test:cookies

**Comment by tinycarol** [comment]
The `CookiesPreAction` unit tests are fixed now 🎉  

There's still a failing test complaining about the version of `portal-kernel`, but running `gradlew baseLine` in that directory does nothing. Any idea how to fix it beyond manually changing the version in the bundle file? @dsanz @jkappler

**Comment by dsanz** [comment]
Looks like a major version change in the `packageinfo` as the result of deleting methods on a `@ProviderType` interface needs an update of `bnd.bnd` file @jkappler, I'm rebasing, then sending an additional commit.

**Comment by dsanz** [comment]
ci:test:cookies

**Comment by jkappler** [comment]
Thanks @dsanz! It must be because of new changes in upstream 😢

**Comment by jkappler** [comment]
@dsanz, I will update

**Comment by dsanz** [comment]
ci:reevaluate:1377401_6339

**Comment by jkappler** [comment]
ci:test:cookies

**Comment by dsanz** [comment]
ci:forward

**Comment by JoyceWang08211** [comment]
ci:test:relevant

**Comment by JoyceWang08211** [comment]
ci:test:relevant

**Comment by gaborlovasdotcom** [comment]
ci:test:relevant

**Comment by dsanz** [comment]
@gaborlovasdotcom can we confirm failing tests are not related to this PR? In that case I'd forward it manually. Thank you :)

**Comment by JoyceWang08211** [comment]
No unique failures are found, please feel free to forward it manually. Thx
cc @dsanz and @gaborlovasdotcom

**Comment by jkappler** [comment]
Sent here: https://github.com/brianchandotcom/liferay-portal/pull/126398

Thanks again @gaborlovasdotcom and @JoyceWang08211.

Amazing work as always @dsanz 👏, hopefully we can get it merged soon!

**Comment by jkappler** [comment]
ci:close

**Pull request closed without merging**